baseband/vlbi_base/base.py

```diff
@@ -48,18 +48,19 @@ class VLBIStreamBase(VLBIFileBase):
 
     _frame_class = None
     _squeezed_shape = None
+    _sample_shape_maker = None
 
-    def __init__(self, fh_raw, header0, sample_shape, bps, complex_data,
-                 thread_ids, samples_per_frame, sample_rate, squeeze=True):
+    def __init__(self, fh_raw, header0, unsliced_shape, bps, complex_data,
+                 subset, samples_per_frame, sample_rate, squeeze=True):
         super(VLBIStreamBase, self).__init__(fh_raw)
         self._header0 = header0
-        self._sample_shape = sample_shape
+        self._unsliced_shape = unsliced_shape
         self._bps = bps
         self._complex_data = complex_data
-        self.thread_ids = thread_ids
         self.samples_per_frame = samples_per_frame
         self.sample_rate = sample_rate
         self.offset = 0
+        self._get_subset_and_sample_shape(subset)
```

Comment:
One might use setters to do this:
```
self.squeeze = squeeze
self.subset = subset
# evalutate lazyproperty
self.sample_shape
```
Without setters
```
self._squeeze = bool(squeeze)
self._subset = subset  # or maybe `self._get_subset(subset)`
self._sample_shape = self._calculate_sample_shape()
```

I think that by looking at both `subset` and `squeeze` in `calculate_sample_shape`, it should be possible to keep integers in subsets for `squeeze=True`.